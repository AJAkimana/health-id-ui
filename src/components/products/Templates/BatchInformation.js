import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import {
  Typography, Grid, Table, TableBody, TableCell, TableHead, TableRow
} from '@material-ui/core';
import { tableStyles } from '../../../assets/styles/products/productDetailStyles';

const BatchInformation = (props) => {
  const {
    classes, renderTableCell, withPriceField,
    skuNumber, manufacturer, currency, moneyFormat, quantityTotal, priceTotal
  } = props;

  return (
    <Fragment>
      <div className={classes.dividerDiv}>
        <Typography paragraph variant="h6" align="left" gutterBottom className={classes.dividerHeaders}>
    Batch Information
        </Typography>
      </div>
      <Grid container spacing={24} className={classes.containerGrid}>
        <Grid item xs={12} style={tableStyles.batchHeader}>
          <Table>
            <TableHead>
              <TableRow style={tableStyles.batchRow}>
                {renderTableCell('left', tableStyles.tableHeader, 'Date Received')}
                {renderTableCell('left', tableStyles.tableHeader, 'SKU')}
                {renderTableCell('left', tableStyles.tableHeader, 'Supplier')}
                {renderTableCell('left', tableStyles.tableHeader, 'Manufacturer')}
                {renderTableCell('left', tableStyles.tableHeader, 'Expiry Date')}
                {renderTableCell('right', tableStyles.tableHeader, 'Quantity')}
                {renderTableCell('right', tableStyles.tableHeader, 'Unit Cost')}
                {renderTableCell('right', tableStyles.tableHeader, 'Total Cost')}
              </TableRow>
            </TableHead>
            <TableBody>
              {withPriceField.map(batch => (
                <TableRow
                  id="outlets-table-row"
                  key={batch.id}
                  style={tableStyles.batchRow}
                >
                  <TableCell component="th" scope="row" style={tableStyles.tableCell}>
                    {batch.dateReceived}
                  </TableCell>
                  <TableCell component="th" scope="row" style={tableStyles.tableCell}>
                    {skuNumber}
                  </TableCell>
                  <TableCell component="th" scope="row" style={tableStyles.tableCell}>
                    {batch.supplier.name}
                  </TableCell>
                  <TableCell component="th" scope="row" style={tableStyles.tableCell}>
                    {manufacturer}
                  </TableCell>
                  <TableCell component="th" scope="row" style={tableStyles.tableCell}>
                    {batch.expiryDate}
                  </TableCell>
                  <TableCell align="right" component="th" scope="row" style={tableStyles.tableCell}>
                    {batch.quantity ? batch.quantity : 0}
                  </TableCell>
                  <TableCell align="right" component="th" scope="row" style={tableStyles.tableCell}>
                    {currency}
                    {' '}
                    {moneyFormat(batch.unitCost)}
                  </TableCell>
                  <TableCell align="right" component="th" scope="row" style={tableStyles.tableCell}>
                    {currency}
                    {' '}
                    {moneyFormat(batch.price)}
                  </TableCell>
                </TableRow>
              ))}
            </TableBody>
            <TableHead>
              <TableRow style={tableStyles.batchRow}>
                <TableCell align="left" style={tableStyles.tableFooter} />
                <TableCell align="left" style={tableStyles.tableFooter} />
                <TableCell align="left" style={tableStyles.tableFooter} />
                <TableCell
                  align="right"
                  style={tableStyles.tableHeader}
                >
                Total
                </TableCell>
                <TableCell align="right" style={tableStyles.tableFooter} />
                <TableCell align="right" style={tableStyles.tableHeader}>
                  {quantityTotal(withPriceField)}
                </TableCell>
                <TableCell
                  align="right"
                  style={tableStyles.tableHeader}
                >
                Grand Total
                </TableCell>
                <TableCell align="right" style={tableStyles.tableHeader}>
                  {currency}
                  {' '}
                  {moneyFormat(priceTotal(withPriceField))}
                </TableCell>
              </TableRow>
            </TableHead>
          </Table>
        </Grid>
      </Grid>
    </Fragment>
  );
};

BatchInformation.propTypes = {
  classes: PropTypes.instanceOf(Object).isRequired,
  renderTableCell: PropTypes.func.isRequired,
  withPriceField: PropTypes.arrayOf(PropTypes.shape({
    map: PropTypes.func
  })).isRequired,
  skuNumber: PropTypes.string.isRequired,
  manufacturer: PropTypes.string.isRequired,
  currency: PropTypes.string.isRequired,
  moneyFormat: PropTypes.func.isRequired,
  quantityTotal: PropTypes.func.isRequired,
  priceTotal: PropTypes.func.isRequired,
};

export default BatchInformation;
