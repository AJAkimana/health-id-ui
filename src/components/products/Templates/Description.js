import React, { Fragment } from 'react';
import {
  Paper, Grid, Card,
  CardActionArea, CardMedia, Chip
} from '@material-ui/core';
import PropTypes from 'prop-types';

const Description = (props) => {
  const {
    classes, renderTextField,
    productCategory, description, tags, image
  } = props;

  return (
    <Fragment>
      <Grid container spacing={24} className={classes.containerGrid}>
        <Grid container item xs={12}>
          <Grid container item xs={8} spacing={24}>
            <Grid item xs={12}>
              <div className={classes.category}>
                {renderTextField(
                  classes.descriptionFields, 'productCategory', 'Category', productCategory.name
                )}
              </div>
              <div>
                {renderTextField(
                  classes.descriptionFields, 'description', 'Description', description
                )}
              </div>

              <Paper elevation={0} className={classes.tagsRoot}>
                {tags.map(tag => (
                  <Chip
                    key={tags.indexOf(tag)}
                    label={tag}
                    className={classes.tagChip}
                    id="tag-chip"
                  />
                ))}
              </Paper>
            </Grid>
          </Grid>

          <Grid item xs={4}>
            <Card elevation={0} className={classes.card}>
              <CardActionArea>
                <CardMedia
                  className={classes.media}
                  image={image}
                  title="Product Image"
                />
              </CardActionArea>
            </Card>
          </Grid>
        </Grid>
      </Grid>
    </Fragment>
  );
};

Description.propTypes = {
  classes: PropTypes.shape({
    containerGrid: PropTypes.string,
    category: PropTypes.string,
    card: PropTypes.string,
    media: PropTypes.string,
    descriptionFields: PropTypes.string,
    name: PropTypes.string,
    tagChip: PropTypes.string,
    tagsRoot: PropTypes.string,
  }).isRequired,
  renderTextField: PropTypes.func.isRequired,
  productCategory: PropTypes.shape({
    name: PropTypes.string
  }).isRequired,
  description: PropTypes.string.isRequired,
  tags: PropTypes.arrayOf(PropTypes.shape({
    map: PropTypes.func,
    indexOf: PropTypes.string
  })).isRequired,
  image: PropTypes.string.isRequired
};

export default Description;
