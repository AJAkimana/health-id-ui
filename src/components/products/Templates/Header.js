import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import BackIcon from '@material-ui/icons/ArrowBack';
import { Link } from 'react-router-dom';
import {
  Typography, Grid, Tooltip, IconButton
} from '@material-ui/core';

const ProductHeader = (props) => {
  const {
    classes, previousPage, productName, children
  } = props;
  return (
    <Fragment>
      <Grid container direction="row" alignItems="center" className={classes.arrowButtonGrid}>
        <Grid item>
          <Link to={previousPage}>
            <Tooltip title="Back to products">
              <IconButton>
                <BackIcon className={classes.arrowIcon} />
              </IconButton>
            </Tooltip>
          </Link>
        </Grid>
        <Grid item>
          <Typography variant="h5" className={classes.arrowButtonLabel}>
            {productName}
          </Typography>
        </Grid>
        <Grid className={classes.buttonGrid}>{children}</Grid>
      </Grid>
    </Fragment>
  );
};
ProductHeader.propTypes = {
  productName: PropTypes.string.isRequired,
  previousPage: PropTypes.string.isRequired,
  classes: PropTypes.instanceOf(Object).isRequired,
  // eslint-disable-next-line react/require-default-props
  children: PropTypes.node,
};

export default ProductHeader;
