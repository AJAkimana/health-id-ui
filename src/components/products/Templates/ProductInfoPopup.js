import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import {
  Paper,

  Grid,
  Chip,
  Tooltip,
  IconButton
} from '@material-ui/core';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import PropTypes from 'prop-types';
import Popper from '@material-ui/core/Popper';
import { ToolbarStyles } from '../../../assets/styles/stock/stock';
import { forwardIcon } from '../productIcons';

export class ProductInfoPopup extends Component {
  state = {
    defaultIcon: 'https://res.cloudinary.com/dojaopytm/image/upload/v1558444184/productPlaceholder.png',
  }

  render() {
    const { defaultIcon } = this.state;
    const {
      row: {
        productName, description, image, tags, id
      },
      position: { x, y },
      classes,
      handleHidePopup,
      onForwardButtonClick
    } = this.props;
    return (
      <div
        id="product_popup_informati"
      >
        <Popper
          style={{
            width: '500px', position: 'absolute', left: `${x}px`, top: `${y}px`
          }}
          open
          transition
          disablePortal
        >
          <Paper>
            <ClickAwayListener onClickAway={handleHidePopup}>
              <Grid
                style={ToolbarStyles.pipmain}
              >
                <Grid xs>
                  <p style={ToolbarStyles.piptitle}>{productName}</p>
                </Grid>
                <Grid
                  container
                  spacing={2}
                >
                  <Grid item xs={image === defaultIcon || image === 'none' ? 11 : 9}>
                    <span
                      style={ToolbarStyles.pipdescription}
                    >
                      {description}
                    </span>
                  </Grid>
                  {
                    (image !== defaultIcon) && (
                      <Grid
                        item
                        xs
                      >
                        <img
                          style={{ width: '100px', height: '100px' }}
                          src={image}
                          alt=""
                        />
                      </Grid>
                    )
                  }
                </Grid>
                <Grid
                  style={{ marginTop: `${image === defaultIcon || image === 'none' ? '20px' : '0px'}` }}
                  container
                  spacing={2}
                >
                  <Grid item xs={10}>
                    <Paper elevation={0} className={classes.tagsRoot}>
                      {tags.map(tag => (
                        <Chip
                          style={ToolbarStyles.piptags}
                          key={tags.indexOf(tag)}
                          label={tag}
                          id="tag-chip"
                        />
                      ))}
                    </Paper>
                  </Grid>
                  <Grid item xs>
                    <Tooltip title="View Product Detail">
                      <IconButton
                        className={classes.iconButton}
                        aria-haspopup="true"
                        onClick={() => {
                          onForwardButtonClick(id);
                        }}
                      >
                        <img
                          alt=""
                          style={ToolbarStyles.pippicon}
                          src={forwardIcon}
                        />
                      </IconButton>
                    </Tooltip>
                  </Grid>
                </Grid>
              </Grid>
            </ClickAwayListener>
          </Paper>

        </Popper>

      </div>
    );
  }
}
ProductInfoPopup.propTypes = {
  row: PropTypes.instanceOf(Object).isRequired,
  classes: PropTypes.instanceOf(Object).isRequired,
  position: PropTypes.instanceOf(Object).isRequired,
  handleHidePopup: PropTypes.func.isRequired,
  onForwardButtonClick: PropTypes.func.isRequired

};
export default withStyles(ToolbarStyles)(ProductInfoPopup);
