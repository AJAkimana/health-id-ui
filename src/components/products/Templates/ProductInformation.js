import React, { Fragment } from 'react';
import { Typography, Grid } from '@material-ui/core';
import PropTypes from 'prop-types';

const ProductInformation = (props) => {
  const {
    classes, renderTextField, measurementUnit, loyaltyWeight, preferredSupplier,
    backupSupplier, id, skuNumber, vatStatus, manufacturer, brand
  } = props;
  return (
    <Fragment>
      <div className={classes.dividerDiv}>
        <Typography
          paragraph
          variant="h6"
          align="left"
          gutterBottom
          className={classes.dividerHeaders}
        >
          Product Information
        </Typography>
      </div>

      <Grid container spacing={24} className={classes.containerGrid}>
        <Grid item xs={4}>
          {renderTextField(
            classes.descriptionFields, 'productId', 'Product ID', id
          )}
        </Grid>
        <Grid item xs={4}>
          {renderTextField(
            classes.descriptionFields, 'skuNumber', 'SKU', skuNumber
          )}
        </Grid>
        <Grid item xs={4}>
          {renderTextField(classes.newTextFields, 'vatStatus', 'VAT Status', vatStatus)}
        </Grid>
        <Grid item xs={4}>
          {renderTextField(
            classes.descriptionFields, 'manufacturer', 'Manufacturer', manufacturer
          )}
        </Grid>
        <Grid item xs={4}>
          {renderTextField(classes.newTextFields, 'brand', 'Brand', brand)}

        </Grid>
        <Grid item xs={4}>
          {renderTextField(
            classes.descriptionFields, 'measurementUnit', 'Measurement Unit',
            measurementUnit.name
          )}
        </Grid>
        <Grid item xs={4}>
          {renderTextField(
            classes.newTextFields, 'loyaltyWeight', 'Loyalty Weight',
            loyaltyWeight
          )}
        </Grid>
        <Grid item xs={4}>
          {renderTextField(
            classes.newTextFields, 'preferredSupplier', 'Preferred Supplier',
            preferredSupplier.name
          )}
        </Grid>
        <Grid item xs={4}>
          {renderTextField(
            classes.newTextFields, 'backupSupplier', 'Backup Supplier',
            backupSupplier.name
          )}
        </Grid>
      </Grid>
    </Fragment>
  );
};

ProductInformation.propTypes = {
  classes: PropTypes.shape({
    containerGrid: PropTypes.string,
    dividerHeaders: PropTypes.string,
    newTextFields: PropTypes.string,
    descriptionFields: PropTypes.string,
    dividerDiv: PropTypes.string,
  }).isRequired,
  renderTextField: PropTypes.func.isRequired,
  measurementUnit: PropTypes.shape({
    name: PropTypes.string
  }).isRequired,
  loyaltyWeight: PropTypes.number.isRequired,
  preferredSupplier: PropTypes.shape({
    name: PropTypes.string
  }).isRequired,
  backupSupplier: PropTypes.shape({
    name: PropTypes.string
  }).isRequired,
  id: PropTypes.string.isRequired,
  skuNumber: PropTypes.string.isRequired,
  vatStatus: PropTypes.bool.isRequired,
  manufacturer: PropTypes.string.isRequired,
  brand: PropTypes.string.isRequired
};

export default ProductInformation;
