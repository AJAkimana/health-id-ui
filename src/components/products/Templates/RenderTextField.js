import React from 'react';
import { TextField } from '@material-ui/core';

const renderTextField = (style, name, label, value) => (
  <TextField
    className={style}
    id={name}
    name={name}
    label={label}
    value={value}
    fullWidth
    InputProps={{ disableUnderline: true, readOnly: true }}
  />
);
export default renderTextField;
