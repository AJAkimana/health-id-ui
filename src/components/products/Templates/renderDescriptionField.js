import React from 'react';
import {
  TextField
} from '@material-ui/core';

const renderDescriptionField = (style, name, label, value) => (
  <TextField
    className={style}
    id={name}
    name={name}
    label={label}
    value={value}
    fullWidth
    InputProps={{ disableUnderline: true, readOnly: true }}
    multiline
    rows={4}
  />
);
export default renderDescriptionField;
