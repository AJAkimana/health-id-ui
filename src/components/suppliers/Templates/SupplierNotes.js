import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import {
  Typography, Grid, Table, TableBody, TableCell, TableHead, TableRow
} from '@material-ui/core';
import { tableStyles } from '../../../assets/styles/suppliers/supplierDetail';

const SupplierNotes = (props) => {
  const {
    classes, renderTableCell, notes
  } = props;

  return (
    <Fragment>
      <div className={classes.dividerDiv}>
        <Typography paragraph variant="h6" align="left" gutterBottom className={classes.dividerHeaders}>
          Supplier Notes
        </Typography>
      </div>
      <Grid container spacing={24} className={classes.tableGrid}>
        {
          (notes && notes.length > 0) ? (
            <Grid item xs={12} style={tableStyles.noteHeader}>
              <Table>
                <TableHead>
                  <TableRow style={tableStyles.noteRow}>
                    {renderTableCell('left', tableStyles.tableHeader, 'Created On')}
                    {renderTableCell('left', tableStyles.tableHeader, 'Message')}
                    {renderTableCell('left', tableStyles.tableHeader, 'Created By')}
                  </TableRow>
                </TableHead>
                <TableBody>
                  {notes && notes.map(note => (
                    <TableRow
                      id="supplier-table-row"
                      key={note.id}
                      style={tableStyles.noteRow}
                    >
                      <TableCell component="th" scope="row" style={tableStyles.tableCell}>
                        {note.createdAt}
                      </TableCell>
                      <TableCell component="th" scope="row" style={tableStyles.tableCell}>
                        {note.note}
                      </TableCell>
                      <TableCell component="th" scope="row" style={tableStyles.tableCell}>
                        {note.supplier.user.firstName}
                        ,
                        {' '}
                        {note.supplier.user.lastName}
                      </TableCell>
                    </TableRow>
                  ))}
                </TableBody>
              </Table>
            </Grid>
          ) : 'No note yet'
        }
      </Grid>
    </Fragment>
  );
};

SupplierNotes.propTypes = {
  classes: PropTypes.instanceOf(Object).isRequired,
  renderTableCell: PropTypes.func.isRequired,
  notes: PropTypes.arrayOf(PropTypes.shape({
    map: PropTypes.func
  })).isRequired,
};

export default SupplierNotes;
