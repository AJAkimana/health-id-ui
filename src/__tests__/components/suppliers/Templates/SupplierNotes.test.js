import React from 'react';
import { shallow, mount } from 'enzyme';

import { SupplierNotes } from '../../../../components/suppliers/Templates/SupplierNotes';

const props = {
  onMouseEnter: jest.fn(),
  renderTableCell: jest.fn(),
  deleteSupplierNote: jest.fn(() =>
    Promise.resolve({
      data: {
        deletedSupplierNote: {
          message: 'Supplier note deleted'
        }
      }
    })
  ),
  supplier: {
    isApproved: false,
    id: 'AlphaX',
    address: '​​​​​NNPC Towers, Central Business District, Abuja.',
    name: 'Damian Inc',
    mobile: '08137519698',
    rating: '5',
    image:
      'https://justcreative.com/wp-content/uploads/2019/09/brand-strategy-workbook-1.jpg',
    email: 'test@gmail.com',
    tier: 'Manufacturer',
    paymentTerms: 'Daily',
    creditDays: '3 days',
    suppliernoteSet: {
      notes: [
        {
          id: 1,
          createdOn: 'Jan 20 2019',
          message: 'Hey bro, nice work',
          createdBy: 'Danilo Silva',
          supplier: { user: { firstName: 'jojo' } }
        },
        {
          id: 2,
          createdOn: 'Jan 20 2019',
          message: 'Hey bro, nice work',
          createdBy: 'Danilo Silva',
          supplier: { user: { firstName: 'jojo' } }
        },
      ]
    },
  },
  addSupplierNote: async () => { },
  renderTextField: jest.fn(),
  variables: {},
  classes: {},
  session: {
    me: {
      role: {
        name: 'Master Admin'
      },
      outlets: [{ id: 1 }],
      activeOutlet: {
        outletpreference: {
          outletTimezone: {
            name: "Africa/Nairobi"
          }
        }
      },
      outlets: [
        {
          id: 'ddfe'
        }
      ]
    }
  },
};

const invalidProps = {
  supplier: {
    isApproved: false,
    id: 'AlphaX',
    address: '​​​​​NNPC Towers, Central Business District, Abuja.',
    name: 'Store Inc',
    mobile: '08137519698',
    rating: '5',
    image:
      'https://justcreative.com/wp-content/uploads/2019/09/brand-strategy-workbook-1.jpg',
    email: 'test@gmail.com',
    tier: 'Manufacturer',
    paymentTerms: 'Daily',
    creditDays: '3 days',
    notes: [
      {
        id: 1,
        createdOn: 'Jan 20 2019',
        message: 'Great Supplier',
        createdBy: 'Danilo Sam'
      },
      {
        id: 2,
        createdOn: 'Jan 20 2019',
        message: 'Great Supplier',
        createdBy: 'Danilo Sam'
      }
    ]
  },
  handleSaveNote: () =>
    new Promise((resolve, reject) =>
      reject({
        graphQLErrors: [
          {
            message: 'error'
          }
        ]
      })
    ),
  handleDelete: () =>
    new Promise((resolve, reject) =>
      reject({
        graphQLErrors: [
          {
            message: 'error'
          }
        ]
      })
    ),
  classes: {
    dividerDiv: 'dividerDiv'
  },
  session: {
    me: {
      role: {
        name: 'Master Admin'
      },
      activeOutlet: {
        outletpreference: {
          outletTimezone: {
            name: 'Africa/Nairobi'
          }
        }
      },
      outlets: [
        {
          id: 'ddfe'
        }
      ]
    }
  },
  addSupplierNote: () =>
  new Promise((resolve, reject) =>
    reject({
      graphQLErrors: [
        {
          message: 'error'
        }
      ]
    })
  ),
deleteSupplierNote: () =>
new Promise((resolve, reject) =>
  reject({
    graphQLErrors: [
      {
        message: 'error'
      }
    ]
  })
),
};

describe('SupplierNotes', () => {
  const wrapper = mount(<SupplierNotes {...props} />)

  const singleNote = {
    id: 8,
    createdOn: 'Jan 20 2019',
    message: 'Great Supplier',
    createdBy: 'Danilo Sam'
  };
  it('should render the SupplierNotes component properly', () => {
    wrapper.instance().handleopenAddModel();
    wrapper.instance().handleopenConfirmationModel();
    wrapper.instance().handleChange({ target: 'efef' });
    wrapper.instance().handleSaveNote();
    wrapper.instance().handleCloseModal();
    wrapper.instance().handleOnRowHover(singleNote, { currentTarget: 'efef' });
    wrapper.instance().handleDelete();
    wrapper.instance().handleMouseLeave();
    expect(typeof wrapper.instance().state).toBe('object');
    expect(wrapper.instance().state.note).toBe(undefined);
    expect(wrapper.instance().state.openAddModel).toBe(false);

  });

  const wrapperInstance = wrapper.instance()
  it('should render "No note yet" with empty notes array', () => {
    wrapper.setProps({ notes: null })
    expect(wrapper.find('Table').length).toBe(0)
  })

  it('should call handleopenAddModel', () => {
    wrapperInstance.handleopenAddModel()
    expect(wrapper.state('openAddModel')).toBeTruthy();
  })

  it('should call handleChange', () => {
    const event = { target: { value: 'momo' } }
    wrapperInstance.handleChange(event)
    expect(wrapper.state('note')).toBe('momo');
  })

  describe('handleSaveNote', () => {
    it('should call "handleCloseModal" on success', () => {
      const spy = jest.spyOn(wrapperInstance, 'handleCloseModal');
      const addSupplierNote = jest.fn().mockResolvedValue(42)
      const refetch = jest.fn();
      wrapper.setProps({ addSupplierNote, refetch })
      wrapperInstance.handleSaveNote()
    })

    it('should call "notify" on failure', () => {
      const addSupplierNote = jest.fn().mockRejectedValue(new Error('Async error'))
      const refetch = jest.fn();
      wrapper.setProps({ addSupplierNote, refetch });
      wrapperInstance.handleSaveNote();
    })
  })

  it('calls handleSaveNote function with error', () => {
    let wrapper = shallow(<SupplierNotes {...invalidProps} />);

    const spy = jest.spyOn(wrapper.instance(), 'handleSaveNote');

    wrapper.instance().handleSaveNote();
    wrapper.update();

    expect(spy).toHaveBeenCalledTimes(1);
  });
  it('calls handleDelete function with error', () => {
    let wrapper = shallow(<SupplierNotes {...invalidProps} />);

    const spy = jest.spyOn(wrapper.instance(), 'handleDelete');

    wrapper.instance().handleDelete();
    wrapper.update();

    expect(spy).toHaveBeenCalledTimes(1);
  });

});
