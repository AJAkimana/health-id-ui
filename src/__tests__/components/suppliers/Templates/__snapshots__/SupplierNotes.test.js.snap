// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplierNotes should render the SupplierNotes component properly 1`] = `
<Fragment>
  <div>
    <WithStyles(Typography)
      align="left"
      gutterBottom={true}
      paragraph={true}
      variant="h6"
    >
      Supplier Notes
    </WithStyles(Typography)>
  </div>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    No note yet
  </WithStyles(Grid)>
</Fragment>
`;
